src/main/scala/wdl4s/cwl/CommandLineTool.scala

```diff
@@ -0,0 +1,97 @@
+package wdl4s.cwl
+
+import eu.timepit.refined._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.string._
+import shapeless.{:+:, CNil}
+import wdl4s.cwl.CwlVersion._
+import wdl4s.cwl.CwlType._
+
+case class CommandLineTool(
+                            inputs: CommandInputParameter :+: Map[CommandInputParameter#Id, CommandInputParameter#`type`] :+: Map[CommandInputParameter#Id, CommandInputParameter] :+: CNil,
+                            outputs: Array[CommandOutputParameter] :+: Map[CommandOutputParameter#Id, CommandOutputParameter#`type`] :+: Map[CommandOutputParameter#Id, CommandOutputParameter] :+: CNil,
+                            `class`: String, //TODO: must be "CommandLineTool" or ?
+                            id: Option[String],
+                            requirements: Option[Array[Requirement]],
+                            hints: Option[Array[String]], //TODO: Any?
+                            label: Option[String],
+                            doc: Option[String],
+                            cwlVersion: Option[CWLVersion],
+                            baseCommand: Option[String :+: Array[String] :+: CNil],
+                            arguments: Option[Array[ECMAScriptExpression :+: CommandLineBinding :+: String :+: CNil]],
+                            stdin: Option[ECMAScriptExpression :+: String :+: CNil],
+                            stderr: Option[ECMAScriptExpression :+: String :+: CNil],
+                            stdout: Option[ECMAScriptExpression :+: String :+: CNil],
+                            successCodes: Option[Array[Int]],
+                            temporaryFailCodes: Option[Array[Int]],
+                            permanentFailCodes: Option[Array[Int]])
+
+case class CommandInputParameter(
+                                  id: Option[String],
+                                  label: Option[String],
+                                  secondaryFiles: Option[Array[ECMAScriptExpression :+: String :+: CNil]],
+                                  format: Option[ECMAScriptExpression :+: Array[String] :+: String :+: CNil], //only valid when type: File
+                                  streamable: Option[Boolean], //only valid when type: File
+                                  doc: Option[String :+: Array[String] :+: CNil],
+                                  inputBinding: Option[CommandLineBinding],
+                                  default: Option[String], //TODO Any type here
+                                  `type`: Option[MyriadInputType]
+
+) {
+  type Id = String
+  type `type` = MyriadCommandInputType
+}
+
+case class CommandInputRecordSchema(
+  `type`: String Refined MatchesRegex[W.`"record"`.T],
```

Comment:
is this saying that this can only be constructed with a String literal that's `"record"`?